modules/okhttp-client/src/main/java/com/spotify/apollo/http/client/HttpClient.java

```diff
@@ -73,12 +73,12 @@ public static HttpClient createUnconfigured() {
       return RequestBody.create(contentType, payload);
     });
 
-    Headers.Builder headersBuilder = new Headers.Builder();
-    apolloRequest.headers().asMap().forEach((k, v) -> headersBuilder.add(k, v));
+    final Headers.Builder headersBuilder = new Headers.Builder();
+    apolloRequest.headers().asMap().forEach(headersBuilder::add);
 
     apolloIncomingRequest
         .flatMap(req -> req.header(AUTHORIZATION_HEADER))
-        .ifPresent(header -> headersBuilder.add(AUTHORIZATION_HEADER, header));
+        .ifPresent(header -> headersBuilder.set(AUTHORIZATION_HEADER, header));
```

Comment:
Is it expected that the header from the incoming request overrides the header explicitly set on the new request? 🤔

Reply:
Not sure how Apollo handles duplicated headers now. Will take a look.

Reply:
It seems it joins multiple headers into a comma separated value: https://github.com/spotify/apollo/blob/master/modules/jetty-http-server/src/main/java/com/spotify/apollo/http/server/ApolloRequestHandler.java#L120

That would be interesting for header like `Authentication`.

Reply:
Yeah, if we're going to send an `Authorization` header I think we want it to have only one value. My question was more about which value should take precedence. Personally I would expect the one explicitly set on the outgoing request to be used and not overridden by the value from the incoming request. Wdyt?

Reply:
Good catch, I also think that what the user sets should take precedence.

Reply:
Agreed, that would make more sense.